packages/codebytes/src/codeByteEditor.tsx

```diff
@@ -80,7 +76,6 @@ export const CodeByteEditor: React.FC<CodeByteEditorProps> = ({
           onChange={(newText: string) => {
             setText(newText);
             onEdit?.(newText, language);
-            const { renderMode } = getOptions();
```

Comment:
just want to make sure i follow... why do we need to check rendermode here? in case we don't want the codebyte to be editable? i'm kinda confused what gets wrapped inside this and what doesn't

Reply:
running this by reach too, to make sure we have full context. it seems like it has to do with editability, but also seems like there are different types of renderModes

Reply:
I'm curious if in speaking to reach we could get more specifics on the possible strings being passed to this and provide it a more specific type other than `string` in `CodeByteEditorProps` so its self documenting.

Reply:
ah, yes! so docs uses a read-only renderMode ("view"), while codebytes uses a "compose" mode. think it's probably more robust to check the value of renderMode here vs just checking if it exists. chris seems to agree.

Reply:
if that's the case, then i think we need to fix what get's wrapped inside the renderMode check in client modules?

Reply:
are view and compose the only two options? if so we could use a boolean

Reply:
chatted a little bit more with chris, but it seems like we don't care too much about the contents of renderMode, just that we want a prop to figure out if we want to fire that tracking event. so i think a boolean is fine, and we just need to pass that from the parent. 

chris checked with britney and they want to keep that tracking event!

Reply:
@zippyzow I'm going to change it to a boolean, so we're just checking if we need to fire tracking on first edit!

Reply:
hmm is there a reason not track the first edit?

Reply:
think this is a britney question, but we can run by her!

Reply:
this just seems like a weird amount of complexity to have? is there any documentation that li wrote up the first time around? im not sure britney would have the context since she wasn't here

Reply:
following up with @christian-dinh on this, can hop on call with him tomorrow to see if this check is really necessary. has more to do with the docs and forums behavior, which, reach would prob have more context on. i was trying to just maintain behavior for now.

Reply:
yeah it's not a huge deal, and we can totally say it's not in scope of this work, it just seems a little odd to me

Reply:
sure! with whatever context you've already obtained from reach